### PR TITLE
sonobuoy integration

### PR DESCRIPTION
**Pull request by yzaccc**
Integrated to run with sonobuoy
- added dockerfile to containerize the project;
- supports to build kube client from incluster kube config;
- added instructions in README to run as a sonobuoy plugin;

to show sonobuoy results:
```
$ make sonobuoy-retrieve
rm -rf results
echo 202201050132_sonobuoy_ab969649-add4-41fb-af36-9e8dfcbce8e9.tar.gz
202201050132_sonobuoy_ab969649-add4-41fb-af36-9e8dfcbce8e9.tar.gz
mkdir results && tar -xf 202201050132_sonobuoy_ab969649-add4-41fb-af36-9e8dfcbce8e9.tar.gz -C results
cat results/plugins/k8s-service-validator-sonobuoy-plugin/*.yaml
name: k8s-service-validator-sonobuoy-plugin
status: passed
meta:
  type: summary
items:
- name: out.json
  status: passed
  meta:
    file: results/global/out.json
```

### CODE REVIEW

File: sonobuoy-plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+sonobuoy-config:
+  driver: Job
+  plugin-name: k8s-service-validator-sonobuoy-plugin
+  result-format: raw
+spec:
+  image: projects-stg.registry.vmware.com/tkgdev/yzac/k8s-svc-validator:latest
```
Comment (knabben) [inline sonobuoy-plugin.yaml:6]: are we  allowed to use the vmware registry here?

File: README.md [hunk 2]
```diff
@@ -89,6 +89,17 @@ $ make build
 $ ./svc-test
 ```
 
+# Run with Sonobuoy
+```
+$ brew instal sonobuoy
```
Comment (knabben) [inline README.md:94]: what if we are not on macosx? can we at least flag this is the macosx command to install it.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM golang:1.16 as build
```
Comment (knabben) [inline Dockerfile:1]: 1.17

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+COPY . /k8s-service-validator
+WORKDIR /k8s-service-validator
+RUN go test -v -c -o svc-test ./tests
+
+FROM docker:dind
+COPY --from=build /k8s-service-validator/svc-test /svc-test
+RUN apk add --update curl && apk add bash && rm -rf /var/cache/apk/*
+
+# kubectl
+RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl
```
Comment (knabben) [inline Dockerfile:11]: is kubectl required?

File: sonobuoy-plugin.yaml (hunk 1 above)
Comment (yzaccc) [reply sonobuoy-plugin.yaml]: changed to use dockerhub

File: README.md (hunk 2 above)
Comment (yzaccc) [reply README.md]: refined the instructions.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+FROM golang:1.16 as build
+COPY . /k8s-service-validator
+WORKDIR /k8s-service-validator
+RUN go test -v -c -o svc-test ./tests
+
+FROM docker:dind
```
[the remark this answers is not in the document]
Comment (yzaccc) [reply Dockerfile]: changed to use debian:stretch-slim

**Review by knabben — APPROVED** [review]

**Pull request merged**